### DISCUSSION TITLE
reset network connection if device is unreachable?

### DISCUSSION THREAD

**Discussion by edgarhildering** [jasonacox/tinytuya#235] [General]
Hi All, I have a weird situation:
- the scan with tinytuya keeps on saying that my device is unreachable (Access rejected by 192.168.xxx.yyy: Network Error: Device Unreachable).
- with Smart Life app I have no trouble switching the device on/off

Q: is there a way to reset the connection with this specific device? All my other switch devices are online and can be switched with tinytuya...

grtz, --Edgar

**Comment by jasonacox** [comment]
Hi @edgarhildering  - what type of device is this?

**Reply by edgarhildering** [reply]
Hi Jason, this is the result from the wizard for the indicated switch:
    {
        "name": "K14 SmartSwich",
        "id": "bfaab80dcd2988ce32eid1",
        "key": "<<...>>",
        "mac": "a0:92:08:fd:ac:30",
        "category": "tdq",
        "product_name": "Aubess Smart Switch 1-gang",
        "product_id": "asinhxudouvo32mi",
        "biz_type": 18,
        "model": "Aubess Smart Switch WZ-cob",
        "sub": false,
        "icon": "https://images.tuyaeu.com/smart/icon/bay1624869296421wDFy/33b52ad8e30c7919bfd1ac69705f80f8.png"
    },

from tuya-raw:
        {
            "active_time": 1667561674,
            "biz_type": 18,
            "category": "tdq",
            "create_time": 1667561674,
            "icon": "smart/icon/bay1624869296421wDFy/33b52ad8e30c7919bfd1ac69705f80f8.png",
            "id": "bfaab80dcd2988ce32eid1",
            "ip": "<<my-extern-id>>",
            "lat": "51.0156",
            "local_key": "42844d594ca17b17",
            "lon": "8.4345",
            "model": "Aubess Smart Switch WZ-cob",
            "name": "K14 SmartSwich",
            "online": true,
            "owner_id": "42495826",
            "product_id": "asinhxudouvo32mi",
            "product_name": "Aubess Smart Switch 1-gang",
            "status": [
                {
                    "code": "switch_1",
                    "value": false
                },
                {
                    "code": "countdown_1",
                    "value": 0
                },
                {
                    "code": "relay_status",
                    "value": "0"
                },
                {
                    "code": "random_time",
                    "value": ""
                },
                {
                    "code": "cycle_time",
                    "value": ""
                },
                {
                    "code": "switch_inching",
                    "value": ""
                }
            ],
            "sub": false,
            "time_zone": "+01:00",
            "uid": "eu1637530536860A1JvF",
            "update_time": 1670885628,
            "uuid": "4c0d556c2ddcc665"
        },

and finally the result of scan:

K14 SmartSwich   Product ID = keyjnuy4s3kre7m7  [Valid payload]:
    Address = 192.168.188.101,  Device ID = bfaab80dcd2988ce32eid1, Local Key = <<...>>,  Version = 3.4, MAC = a0:92:08:fd:ac:30
    Access rejected by 192.168.188.101: Network Error: Device Unreachable

I could take the power off and restart the switch, but the switch is (physically!) hard to reach: it is in the attic, where the roof hits the floor. Furthermore, if this happens (out of the blue) it will happen again over time. so its not a permanent solution.

This switch by the way is the only switch I have with version 3.4. 

grtz, --Edgar

**Reply by uzlonewolf** [reply]
Do you know what circuit it's connected to?  Instead of climbing up there you could just shut off the fuse or circuit breaker for a few seconds to reboot it.

Are you running TinyTuya somewhere or is the Smart Life app open?  Most devices have a limit of 1-3 simultaneous open connections and will refuse additional connections until one of the open ones is closed.

Lastly, can the Smart Life app see and control it when it's doing this, or is it offline completely?

**Reply by edgarhildering** [reply]
Sorry, uzlonewolf, but the fuse is not an option: the router, home automation and some other lightweight stuff is connected to this circuit.
The smart life app is not open or active on my phone. There are no other tinytuya instances running except the 'python3 -m tinytuya scan' program.
And yes, the Smart Life app can see and control the switch (i.e. I can turn on/off the light with the smart life app)

So that is the weird thing. I am pretty sure there is some software condition blocking access... I am willing to dive in the code, although i have no clu where to begin. On top of that: it has worked for a couple of days... And I really did not change anything.

(I have a solid IT background as a retired software architect with more than 20 years on the counter. And when troubleshooting, these were always our customer's last famous last words: No, I didn't change anything! :smile:  But truly, i didn't...)

**Reply by uzlonewolf** [reply]
Unfortunately in that case I have nothing besides the usual "check to make sure the device firmware is the latest (in the Smart Life app) and try turning it off and back on."  There is nothing in TinyTuya that could cause this, it's entirely the device, network, or PC (is the firewall on your PC blocking access?).  If it's not your network or PC then the only thing you can do is replace the device.

**Reply by edgarhildering** [reply]
I might get up to the attic, tomorrow to interrupt the power. As you can deduct from my lat, lon, it is pretty cold right here at this moment. 
I could replace this switch with a version 3.3, i still have in the cubboard. And do some more testing with the 'recovered' switch...

stay tuned...

**Reply by edgarhildering** [reply]
ok, I braved the cold and went to the attic. I decided not to swap the switch but instead perfrom a reset of the switch by pressing the button +10 seconds. Then added it again in the Smart Life app. That yielded the same data in the app, with the exception of the key, which was refreshed.

- result of the wizard: success, no more network error.
- result of the scanner: success, no more network error.
- I replaced the key in my own software and then everything worked again...

imho, two things may have changed:

1. the reset also reset the internal firmware. This makes connection with tinytuya possible again. This is difficult to investigate because a connection based on the old key is no longer possible...
2. the old key no longer worked in tinytuya (but still worked in the cloud!). The new key works in both environments. This is also quite difficult to investigate. Because that sounds like a crypto problem...

For now, the problem has been solved, but the underlying cause has not yet been found. Of course I keep an eye on things. If this problem occurs again, I will report back.

thanks so far, and greetings from snowy Germany, --Edgar
